helix-term/src/ui/prompt.rs

```diff
@@ -205,61 +205,61 @@ impl Prompt {
         self.cursor = self.line.len();
     }
 
-    pub fn delete_char_backwards(&mut self) {
+    pub fn delete_char_backwards(&mut self, cx: &Context) {
         let pos = self.eval_movement(Movement::BackwardChar(1));
         self.line.replace_range(pos..self.cursor, "");
         self.cursor = pos;
 
         self.exit_selection();
-        self.completion = (self.completion_fn)(&self.line);
+        self.completion = (self.completion_fn)(&self.line, cx);
```

Comment:
Rather than calling `completion_fn` in every function and having to pass the context through, we could call it inside `handle_event` before returning? It might be a bit excessive if no recomputation is needed (on move_) though.

Reply:
IMHO, this is fine. Not everything needs a completion.

That said, if you really want me to make the change, it should be easily doable.